.github/workflows/release.yml

```diff
@@ -3,14 +3,82 @@ name: Release
 on:
   workflow_dispatch:
 
+
 jobs:
-  gradle:
+  # TOODO how do I ensure that nothing is running on a different commit
+  # 1. We look at the target branch and figure out whether this should be a BUILD
+  # release or a patch release
+  get-update-type:
+    # If the context.ref is refs/heads/main we want to do a standard release
+    # If the context.ref is refs/heads/* we want to do a patch release
+    # If the context.ref is some other ref, we want to fail
+    # Selecting a tag github.ref becomes e.g.: refs/tags/4.1.9.0
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    outputs: # This means this can later be referenced with needs.get-update-type.outputs.update-type
+      update-type: ${{ steps.update-type.outputs.result }}
+    steps:
+      - name: Calculate update type
+        uses: actions/github-script@v7
+        id: update-type
+        with:
+          script: |
+            if (context.ref == "refs/heads/main") {
+                return "BUILD";
+            } else if (context.ref.startsWith("refs/heads/")) {
+                return "PATCH";
+            } else {
+                throw new Error("Target must be a patch branch or main, but was: " + context.ref);
+            }
+          result-encoding: string
+      - name: Print update type
+        shell: bash
+        run: echo "${{steps.update-type.outputs.result}}"
+
+  # 2. In parallel, we run:
+  #   a) all the tests
+  #   b) the mixed-mode tests
+  test:
+    needs: [get-update-type]
```

Comment:
Does this actually need `get-update-type`?

Reply:
No it does not, removed.